tensorflow_addons/layers/normalizations.py

```diff
@@ -321,3 +321,207 @@ def __init__(self, **kwargs):
+            self._add_eps_learned_weight()
+        self.built = True
+        super().build(input_shape)
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+    def get_config(self):
+        config = {
+            "axis": self.axis,
+            "epsilon": self.epsilon,
+            "learned_epsilon": self.use_eps_learned,
+            "beta_initializer": tf.keras.initializers.serialize(self.beta_initializer),
+            "gamma_initializer": tf.keras.initializers.serialize(
+                self.gamma_initializer
+            ),
+            "eps_learned_initializer": tf.keras.initializers.serialize(
+                self.eps_learned_initializer
+            ),
+            "beta_regularizer": tf.keras.regularizers.serialize(self.beta_regularizer),
+            "gamma_regularizer": tf.keras.regularizers.serialize(
+                self.gamma_regularizer
+            ),
+            "beta_constraint": tf.keras.constraints.serialize(self.beta_constraint),
+            "gamma_constraint": tf.keras.constraints.serialize(self.gamma_constraint),
+        }
+        base_config = super().get_config()
+        return dict(**base_config, **config)
+
+    def call(self, inputs):
+        epsilon = tf.math.abs(self.epsilon)
+        if self.use_eps_learned:
+            epsilon += self.eps_learned
+        nu2 = tf.reduce_mean(tf.square(inputs), axis=self.axis, keepdims=True)
+        inputs *= tf.math.rsqrt(nu2 + epsilon)
+        return self.gamma * inputs + self.beta
+
+    def _create_input_spec(self, input_shape):
+        ndims = len(tf.TensorShape(input_shape))
+        for idx, x in enumerate(self.axis):
+            if x < 0:
+                self.axis[idx] = ndims + x
+
+        # Validate axes
+        for x in self.axis:
+            if x < 0 or x >= ndims:
+                raise ValueError("Invalid axis: %d" % x)
+
+        if len(self.axis) != len(set(self.axis)):
+            raise ValueError("Duplicate axis: %s" % self.axis)
+
+        axis_to_dim = {x: input_shape[x] for x in self.axis}
+        self.input_spec = tf.keras.layers.InputSpec(ndim=ndims, axes=axis_to_dim)
+
+    def _check_axis(self):
+        if self.axis == 0:
+            raise ValueError(
+                "You are trying to normalize your batch axis. You may want to "
+                "use tf.layer.batch_normalization instead"
+            )
+        if self.axis == -1:
+            raise ValueError(
+                "You are trying to normalize your channel axis. You may want to "
+                "use GroupNormalization layer instead from tf_addons"
+            )
+        if self.axis != [1, 2]:
+            raise ValueError(
+                "FilterResponseNormalization operates on per-channel basis."
+                "Axis values should be spatial dimensions"
+            )
+
+    def _check_if_input_shape_is_none(self, input_shape):
+        dim1, dim2 = input_shape[self.axis[0]], input_shape[self.axis[1]]
```

Comment:
This assumes that `len(self.axis) == 2`, but that isn't asserted anywhere. Also, what about when `axis` is an int?

Reply:
Will resolve this

Reply:
Same case. When checking for axis values, we are making sure that it's a list for now